src/lib.rs

```diff
@@ -8,20 +8,21 @@
 //! extern crate rsa;
 //! extern crate rand;
 //!
-//! use rsa::{PublicKey, RSAPrivateKey, PaddingScheme};
+//! use rsa::{PublicKey, RSAPrivateKey, RSAPublicKey, PaddingScheme};
 //! use rand::rngs::OsRng;
 //!
 //! let mut rng = OsRng;
 //! let bits = 2048;
-//! let key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
+//! let private_key = RSAPrivateKey::new(&mut rng, bits).expect("failed to generate a key");
+//! let public_key = RSAPublicKey::from(private_key.clone());
```

Comment:
This should probably work without calling `clone`, as this means duplicating a bunch of secrets, when we actually only want the public parts